core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/transmit/event_provider.go

```diff
@@ -61,7 +60,9 @@ func NewTransmitEventProvider(
 	if err != nil {
 		return nil, err
 	}
-	err = logPoller.RegisterFilter(logpoller.Filter{
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	err = logPoller.RegisterFilter(ctx, logpoller.Filter{
```

Comment:
You can pass caller's context here as well.

Reply:
Sweet